src/main/java/mate/academy/springbootwebgreqit/controller/CategoryController.java

```diff
@@ -0,0 +1,51 @@
+package mate.academy.springbootwebgreqit.controller;
+
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import mate.academy.springbootwebgreqit.dto.BookDtoWithoutCategotyIds;
+import mate.academy.springbootwebgreqit.dto.category.CategoryDto;
+import mate.academy.springbootwebgreqit.service.CategoryService;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/categories")
+public class CategoryController {
+    private final CategoryService categoryService;
+
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @PostMapping
+    public CategoryDto createCategory(@Valid @RequestBody CategoryDto categoryDto){
+        return categoryService.save(categoryDto);
+    }
+
+    @GetMapping
```

Comment:
Should we add `@PreAuthorize` or you have other opinion?

Reply:
yeah, we can add @PreAuthorize with User role

Reply:
hmmm, i think it's ok when user can find all books

Reply:
yes, we should add `@PreAuthorize`